### PR TITLE
Gist

### PR DESCRIPTION
**Pull request by danacastroman**

### CODE REVIEW

File: ejercicio1.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  description = gets.chomp
+  puts "public: "
+  public = gets.chomp
+
+  uri = URI.parse('https://api.github.com/gists')
+
+
+  data = {
+            'description' => "#{description}",
+            'public' => "#{public}",
```
Comment (pazcuturi) [inline ejercicio1.rb]: Shouldn't `public` be a boolean?

File: ejercicio1.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+begin
+  puts "nombre del archivo: "
+  namefile = gets.chomp
+  content = File.read(namefile)
+
+  puts "descripcion: "
+  description = gets.chomp
+  puts "public: "
+  public = gets.chomp
```
Comment (pazcuturi) [inline ejercicio1.rb]: Should we make sure that the value is either `true` or `false`?

File: ejercicio1.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  request.body = data.to_json
+  response = http.request(request)
+
+  json = response.body
+  url = JSON.parse(json)
+
+  puts url['url']
+  
+rescue => exception
+  p exception.message
```
Comment (pazcuturi) [inline ejercicio1.rb]: What do you think about showing a more user-friendly error message according to each case? e.g. file not found, connection error, etc.

File: ejercicio1.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  if public == 'si'
+    public = true
+  elsif public == 'no'
+    public = false
+  else
+    puts "respuesta incorrecta"
+  end
+
+  uri = URI.parse('https://api.github.com/gists')
+
```
Comment (pazcuturi) [inline ejercicio1.rb]: Can we remove extra blank lines? Here and in other places of the file

File: ejercicio1.rb [hunk 5]
```diff
@@ -4,15 +4,14 @@
 require 'httparty'
 
 begin
-  puts "nombre del archivo: "
+  puts "ingrese ruta dal archivo: "
```
Comment (Joelit0) [inline ejercicio1.rb]: Exercise 1 doesn't need a path, it needs a file

File: ejercicio1.rb [hunk 6]
```diff
@@ -33,27 +30,24 @@
...
-        
 
   http = Net::HTTP.new(uri.host, uri.port)
   http.use_ssl = true
   http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
   request = Net::HTTP::Post.new(uri.request_uri)
-  request['Authorization'] = 'token c90f2da8cea3cef00e96e96bc1433d7024479715'
-
+  request['Authorization'] = 'token bb3f0a45790a8b18bf1468c70957c55293e7fe32'
```
Comment (Joelit0) [inline ejercicio1.rb]: You shouldn't expose your Github token, use 'dotenv' gem to hide this information.

File: ejercicio1.rb [hunk 7]
```diff
@@ -22,8 +21,6 @@
   end
 
   uri = URI.parse('https://api.github.com/gists')
```
Comment (Joelit0) [inline ejercicio1.rb]: Maybe you should save the link to a constant because the link should not change.

File: ejercicio1.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  description = gets.chomp
+  puts "Quieres el gist publico? si/no"
+  public = gets.chomp
+  
+  if public == 'si'
+    public = true
+  elsif public == 'no'
+    public = false
+  else
+    puts "respuesta incorrecta"
```
Comment (Joelit0) [inline ejercicio1.rb]: Check what happens if the program enters here. What would be the value of "public"? Is the request made?

File: ejercicio1.rb [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  uri = URI.parse('https://api.github.com/gists')
+  data = {
+            'description' => "#{description}",
```
Comment (Joelit0) [inline ejercicio1.rb]: Is necessary use interpolation here?  I think not because the variable is already a string.

File: main.rb [hunk 10]
```diff
@@ -0,0 +1,45 @@
+require_relative 'request'
+require_relative 'gist_content'
+
+loop do
+  puts "Ingrese nombre del archivo "
```
Comment (pazcuturi) [inline main.rb]: Remove empty space please

File: main.rb [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      @state = false
+      break
+    else
+      puts "Respuesta incorrecta"
+    end
+  end
+
+  values = Received_values.new(namefile, description, @state)
+  values.file_existence
+  request =  Assemble.new(namefile, description, @state, values.content)
```
Comment (pazcuturi) [inline main.rb]: What happens with the request if the filename doesn't exist? Does it get made either way?

File: main.rb [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+      @state = false
+      break
+    else
+      puts "Respuesta incorrecta"
+    end
+  end
+
+  values = Received_values.new(namefile, description, @state)
+  values.file_existence
+  if values.content != nil
```
Comment (pazcuturi) [inline main.rb]: Add an empty line please. Also, do we need to compare against nil?

File: main.rb [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+    request =  Assemble.new(namefile, description, @state, values.content)
```
Comment (pazcuturi) [inline main.rb]: Remove extra space please

**Review by pazcuturi — APPROVED** [review]

**Pull request merged**